### PR TITLE
CLOUDP-95673: Quickstart cluster creation confirmation

### PR DESCRIPTION
**Pull request by colm-quinn**
## Proposed changes
* Confirm before makign a cluster with quickstart
* Add summary of changes before creating cluster

_Jira ticket:_ CLOUDP-95673

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments
Command:
```
bin/mongocli atlas quickstart --profile=atlas-dev --tier M0
You are creating a new Atlas cluster and enabling access to it.

Press [Enter] to use the default values.

Enter [?] on any option to get help.

[Set up your Atlas cluster]
? Cluster Name [This can't be changed later] Quickstart-1632214919
? Cloud Provider AWS
? Cloud Provider Region US_EAST_1
? Do you want to load sample data into Quickstart-1632214919? No

[Set up your database authentication access details]
? Database User Username Quickstart-1632214919
? Database User Password [Press Enter to use an auto-generated password '<REDACTED>'] <REDACTED>

[Set up your database network access details]
? Access List Entry [Press Enter to use your public IP address '109.255.179.192'] 1.2.3.4,5.6.7.8

[Connect to Quickstart-1632155644]
? Do you want to access Quickstart-1632155644 with MongoDB Shell? No

[Summary of changes]
Cluster Name:				Quickstart-1632214919
Cluster Tier:				M0
Cloud Provider:				AWS
Region:					US_EAST_1
Cluster Disk Size (GiB):		0.5
Database Username:			Quickstart-1632214919
Allow connections from (IP Address):	1.2.3.4, 5.6.7.8
Load sample data:			No
Open shell:				No
? Do you want to create a new cluster Quickstart-1632155644 with the following settings? No
Error: user-aborted. Not creating cluster
```

Select create cluster:
```
? Do you want to create a new cluster Quickstart-1632155677 with the following settings? Yes
We are deploying Quickstart-1632155677...
Creating your cluster... [It's safe to 'Ctrl + C']
\
```
Cluster being created:
image

### CODE REVIEW

File: internal/cli/atlas/quickstart/confirm_cluster_setup.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+func (opts *Opts) askConfirmConfigQuestion() error {
+	if opts.Confirm {
+		return nil
+	}
+
+	diskSize := uint64(0)
+	if opts.newCluster().DiskSizeGB != nil {
+		diskSize = uint64(*opts.newCluster().DiskSizeGB)
+	}
```
Comment (colm-quinn) [inline internal/cli/atlas/quickstart/confirm_cluster_setup.go:32]: M0 comes in as nil, so setting to 0 in those instances

File: internal/cli/atlas/quickstart/confirm_cluster_setup.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"errors"
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+func (opts *Opts) askConfirmConfigQuestion() error {
+	if opts.Confirm {
+		return nil
```
Comment (fmenezes) [inline internal/cli/atlas/quickstart/confirm_cluster_setup.go:26]: [q] when is this condition reached? do we have a flag?

File: internal/cli/atlas/quickstart/confirm_cluster_setup.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:32]: M0 is 0.5GiB

Comment (colm-quinn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:32]: Cool, I'll update so the text shows that in this instance, thanks!

File: internal/cli/atlas/quickstart/confirm_cluster_setup.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	fmt.Printf(`
+[Summary of changes]
+Name:			%s
+Tier:			%s
+Disk Size (GB):		%d
+User:			%s
+Ip Address:		%s
+`,
```
Comment (gssbzn) [inline internal/cli/atlas/quickstart/confirm_cluster_setup.go:41]:
```suggestion
Cluster Name:			%s
Cluster Tier:			%s
Cluster Disk Size (GB):		%d
Database Username:			%s
Allowed connections from (IP Address):		%s
`,
```

And can we please add Cloud provider and region?

Comment (colm-quinn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:41]: Sure, anything else we might need? (above was what was in the ticket, not sure what's important to our users)

Comment (gssbzn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:41]: also super nit `opts.IPAddresses` is a slice and why it adds brackets on your example, wondering if we should format it better, `strings.Join` kind of thing

Comment (colm-quinn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:41]: I actually quite like the output with the brackets (it felt like the right representation to me): 
```
User:			Quickstart-1632155644
Ip Address:		[109.255.179.192]
```

Comment (gssbzn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:41]: > anything else we might need? 

main feedback was tier and disk size as it's "hidden" but based on personal experience with a different tool all things a user can pick should be shown to give them/us a change to double check those are the right settings

so maybe a yes/no to the `load sample data` and open shell questions?

Comment (colm-quinn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go]: Settled on the `strings.join` format without the brackets

File: internal/cli/atlas/quickstart/confirm_cluster_setup.go (hunk 2 above)
Comment (fmenezes) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:27]: my point here is, are going to add a flag to the command? like `--force` or `--confirm`?

Comment (colm-quinn) [reply internal/cli/atlas/quickstart/confirm_cluster_setup.go:27]: Got it. Let me add that quickly!

**Pull request merged**